File: .dev/python-impl/slices.py
import sys, csv
from os.path import basename, splitext, dirname

loop_marks_path = sys.argv[1]
hz = int(sys.argv[2]) if len(sys.argv) > 2 else 44100

fname = splitext(basename(loop_marks_path))[0]
sample_name = fname + ".wav"
code = 35

group_template = """<group> lovel={level_min} hivel={level_max} sample={sample_path} loop_mode=loop_continuous
"""
patch_template = """<region> key={code} offset={offset} end={end}	loop_start={offset} loop_end={end}"""

with open(dirname(loop_marks_path) + "/" + fname + ".sfz", "w") as fout:
	with open(loop_marks_path, newline="") as fmarks:
		leftp = None
		first = True
		for row in csv.reader(fmarks, delimiter="\t"):
			p = round(float(row[0].replace(",", ".")) * hz)
			if leftp == None:
				leftp = p
				continue
			rightp = p
			#print((leftp, rightp))
			data = {
				"code": code,
				"level_min": 0,
				"level_max": 127,
				"sample_path": sample_name,
				"offset": leftp,
				"end": rightp
			}
			region = (group_template.format(**data) if first else "") + patch_template.format(**data)
			first = False
			fout.write(region + "\n")
			code += 1
			leftp = rightp

File: .dev/golang-impl/slices.go
package main

import (
	"os"
	"fmt"
	"strconv"
	"path/filepath"
	"strings"
	"text/template"
	"encoding/csv"
	"bufio"
	"io"
	"math"
)

func check(e error) {
  if e != nil {
		panic(e)
  }
}

func main() {
	loopMarksPath := os.Args[1]
	hz := 44100
	if len(os.Args) > 2 && os.Args[2] != "-" {
		var err error
		hz, err = strconv.Atoi(os.Args[2])
		check(err)
	}

	regionsType := 0
	if len(os.Args) > 3 {
		var err error
		regionsType, err = strconv.Atoi(os.Args[3])
		check(err)
	}

	fmt.Println(regionsType);
	fmt.Println(hz)
	fmt.Println(loopMarksPath)

	dirname, fname := filepath.Split(loopMarksPath)
	fname = strings.TrimSuffix(fname, filepath.Ext(fname))
	sampleName := fname + ".wav"

	//fmt.Println(dirname)
	//fmt.Println(fname)
	//fmt.Println(sampleName)

	groupTemplate := template.Must(template.New("group").Parse(`
<group> lovel={{.LevelMin}} hivel={{.LevelMax}} sample={{.SamplePath}} loop_mode=loop_continuous
`))
	patchTemplate := template.Must(template.New("region").Parse(
`<region> key={{.Code}} offset={{.Offset}} end={{.End}} loop_start={{.Offset}} loop_end={{.End}}
`))

	fout, err := os.Create(dirname + "/" + fname + ".sfz")
	check(err)
	defer fout.Close()

	fmarks, err := os.Open(loopMarksPath)
	check(err)
	defer fmarks.Close()

	i := 1
	code := 35
	leftp := -1
	rightp := 0
	first := true
	r := csv.NewReader(bufio.NewReader(fmarks))
	r.Comma = '\t'
	//r.Comment = '#'
	for {
		row, err := r.Read()
		if err == io.EOF {
			break
		}
		check(err)

		if regionsType == 0 {
			p, _ := strconv.ParseFloat(strings.Replace(row[0], ",", ".", -1), 64)
			p = math.Floor(p * float64(hz))

			if leftp == -1 {
				leftp = int(p)
				continue
			}
			rightp = int(p)
		} else {
			fmt.Println(i)
			fmt.Println(row)

			p, _ := strconv.ParseFloat(strings.Replace(row[0], ",", ".", -1), 64)
			p = math.Floor(p * float64(hz))
			leftp = int(p)

			p2, _ := strconv.ParseFloat(strings.Replace(row[1], ",", ".", -1), 64)
			p2 = math.Floor(p2 * float64(hz))
			rightp = int(p2)
		}

		data := struct {
      Code int
      LevelMin int
      LevelMax int
      SamplePath string
      Offset int
      End int
    } { code, 0, 127, sampleName, leftp, rightp }
		if first {
			groupTemplate.Execute(fout, data)
		}
		patchTemplate.Execute(fout, data)

		fmt.Println(leftp, rightp)

		first = false
		code += 1
		leftp = rightp
		i++
	}
}
